src/Microsoft.Health.SqlServer.UnitTests/ManagedIdentitySqlConnectionTests.cs

```diff
@@ -18,47 +21,62 @@ public class ManagedIdentitySqlConnectionTests
-    }
 
-    [Fact]
-    public async Task GivenManagedIdentityConnectionType_WhenSqlConnectionRequested_AccessTokenIsSet()
+    [Theory]
+    [InlineData(SqlServerAuthenticationType.ManagedIdentity)]
+    [InlineData(SqlServerAuthenticationType.WorkloadIdentity)]
+    public async Task GivenManagedIdentityConnectionType_WhenSqlConnectionRequested_AccessTokenIsSet(SqlServerAuthenticationType sqlServerAuthenticationType)
     {
+        InitializeTest(sqlServerAuthenticationType);
+
         SqlConnection sqlConnection = await _sqlConnectionFactory.GetSqlConnectionAsync().ConfigureAwait(false);
 
         Assert.Equal(TestAccessToken, sqlConnection.AccessToken);
     }
 
-    [Fact]
-    public async Task GivenDefaultConnectionType_WhenSqlConnectionRequested_DatabaseIsSet()
+    [Theory]
+    [InlineData(SqlServerAuthenticationType.ManagedIdentity)]
+    [InlineData(SqlServerAuthenticationType.WorkloadIdentity)]
+    public async Task GivenDefaultConnectionType_WhenSqlConnectionRequested_DatabaseIsSet(SqlServerAuthenticationType sqlServerAuthenticationType)
     {
+        InitializeTest(sqlServerAuthenticationType);
         SqlConnection sqlConnection = await _sqlConnectionFactory.GetSqlConnectionAsync().ConfigureAwait(false);
 
         Assert.Equal(DatabaseName, sqlConnection.Database);
     }
 
-    [Fact]
-    public async Task GivenDefaultConnectionType_WhenSqlConnectionToMasterRequested_MasterDatabaseIsSet()
+    [Theory]
+    [InlineData(SqlServerAuthenticationType.ManagedIdentity)]
+    [InlineData(SqlServerAuthenticationType.WorkloadIdentity)]
+    public async Task GivenDefaultConnectionType_WhenSqlConnectionToMasterRequested_MasterDatabaseIsSet(SqlServerAuthenticationType sqlServerAuthenticationType)
     {
+        InitializeTest(sqlServerAuthenticationType);
         SqlConnection sqlConnection = await _sqlConnectionFactory.GetSqlConnectionAsync(MasterDatabase).ConfigureAwait(false);
 
         Assert.Equal(MasterDatabase, sqlConnection.Database);
     }
+
+    private void InitializeTest(SqlServerAuthenticationType sqlServerAuthenticationType)
+    {
+        SqlServerDataStoreConfiguration sqlServerDataStoreConfiguration = new SqlServerDataStoreConfiguration
+        {
+            ConnectionString = $"Server={ServerName};Database={DatabaseName};",
+            AuthenticationType = sqlServerAuthenticationType,
+        };
+
+        TokenCredential tokenCredential = Substitute.For<TokenCredential>();
+
+        AccessToken token = new AccessToken(TestAccessToken, DateTime.Now.AddHours(5));
+
+        tokenCredential.GetToken(new TokenRequestContext(new[] { _azureResource }), CancellationToken.None).Returns(token);
+
+        var sqlConfigOptions = Options.Create(sqlServerDataStoreConfiguration);
+        _sqlConnectionFactory = new ManagedIdentitySqlConnectionBuilder(
+            new DefaultSqlConnectionStringProvider(sqlConfigOptions), SqlConfigurableRetryFactory.CreateNoneRetryProvider(), tokenCredential);
```

Comment:
Presumably the `DefaultSqlConnectionStringProvider` should be mocked via `NSubstitue` as well, right?

Reply:
This value is still needed in tests that verifies the database name.